### PR TITLE
Add http propagation APIs.

### PR DESCRIPTION
**Pull request by HailongWen**
* Add an API in `PropagationComponent` for B3 format access.

* Add contrib-http-util, with initial SD format support.

* This is part of the https://github.com/census-instrumentation/opencensus-java/issues/908

### CODE REVIEW

File: contrib/http_util/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+working with HTTP.
+
+## Quickstart
+
+### Add the dependencies to your project
+
+For Maven add to your `pom.xml`:
+```xml
+<dependencies>
+  <dependency>
```
Comment (bogdandrutu) [inline contrib/http_util/README.md]: To be correct you need a dependency on the opencensus-api as well

File: contrib/http_util/src/main/java/io/opencensus/contrib/http/util/HttpPropagationUtil.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import io.opencensus.trace.propagation.TextFormat;
+
+/** Utility class to get all supported {@link TextFormat}. */
+public class HttpPropagationUtil {
+  /**
+   * Returns the Stack Driver format.
+   *
+   * @return the Stack Driver format.
+   */
+  public static TextFormat cloudTraceFormat() {
```
Comment (bogdandrutu) [inline contrib/http_util/src/main/java/io/opencensus/contrib/http/util/HttpPropagationUtil.java]: getCloudTraceFormat

File: contrib/http_util/src/test/java/io/opencensus/contrib/http/util/HttpPropagationUtilTest.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import org.junit.runners.JUnit4;
+
+/** Unit tests for {@link HttpPropagationUtil}. */
+@RunWith(JUnit4.class)
+public class HttpPropagationUtilTest {
+
+  @Test
+  public void cloudTraceFormatNotNull() {
+    TextFormat cloudTraceFormat = HttpPropagationUtil.cloudTraceFormat();
+    assertThat(cloudTraceFormat).isNotNull();
```
Comment (bogdandrutu) [inline contrib/http_util/src/test/java/io/opencensus/contrib/http/util/HttpPropagationUtilTest.java]: test that is instance of CloudTraceFormat

File: impl_core/src/main/java/io/opencensus/implcore/trace/propagation/PropagationComponentImpl.java [hunk 4]
```diff
@@ -18,13 +18,20 @@
...
+  private final B3Format b3Format = new B3Format();
 
   @Override
   public BinaryFormat getBinaryFormat() {
     return binaryFormat;
   }
+
+  @Override
+  public TextFormat getB3Format() {
+    return b3Format;
```
Comment (bogdandrutu) [inline impl_core/src/main/java/io/opencensus/implcore/trace/propagation/PropagationComponentImpl.java]: Add a test that you return the right instance class.

File: contrib/http_util/README.md (hunk 1 above)
Comment (HailongWen) [reply contrib/http_util/README.md]: Done.

File: contrib/http_util/src/main/java/io/opencensus/contrib/http/util/HttpPropagationUtil.java (hunk 2 above)
Comment (HailongWen) [reply contrib/http_util/src/main/java/io/opencensus/contrib/http/util/HttpPropagationUtil.java]: Done.

File: contrib/http_util/src/test/java/io/opencensus/contrib/http/util/HttpPropagationUtilTest.java (hunk 3 above)
Comment (HailongWen) [reply contrib/http_util/src/test/java/io/opencensus/contrib/http/util/HttpPropagationUtilTest.java]: Done.

File: impl_core/src/main/java/io/opencensus/implcore/trace/propagation/PropagationComponentImpl.java (hunk 4 above)
Comment (HailongWen) [reply impl_core/src/main/java/io/opencensus/implcore/trace/propagation/PropagationComponentImpl.java]: Done.

File: contrib/http_util/src/test/java/io/opencensus/contrib/http/util/HttpPropagationUtilTest.java (hunk 3 above)
Comment (bogdandrutu) [reply contrib/http_util/src/test/java/io/opencensus/contrib/http/util/HttpPropagationUtilTest.java]: assertThat(cloudTraceFormat).isInstanceOf(CloudTraceFormat.class);

Same everywhere.

File: api/src/main/java/io/opencensus/trace/propagation/PropagationComponent.java [hunk 5]
```diff
@@ -32,6 +32,14 @@
    */
   public abstract BinaryFormat getBinaryFormat();
 
+  /**
+   * Returns the {@link TextFormat} with the provided implementations. If no implementation is
```
Comment (bogdandrutu) [inline api/src/main/java/io/opencensus/trace/propagation/PropagationComponent.java]: Please provide a link to the documentation of B3 and get some background about it:
https://github.com/openzipkin/b3-propagation

Comment (HailongWen) [reply api/src/main/java/io/opencensus/trace/propagation/PropagationComponent.java]: Done.
Besides, for `HttpPropagationUtil`, I also added a link to the CloudTraceFormat. To make the javadoc available I also make this class public. Please let me know if this should not be done.

File: contrib/http_util/src/main/java/io/opencensus/contrib/http/util/HttpPropagationUtil.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+ * Utility class to get all supported {@link TextFormat}.
+ *
+ * @since 0.11.0
+ */
+public class HttpPropagationUtil {
+
+  private HttpPropagationUtil() {}
+
+  /**
+   * Returns the Stack Driver format. See {@link CloudTraceFormat} for more information.
```
Comment (bogdandrutu) [inline contrib/http_util/src/main/java/io/opencensus/contrib/http/util/HttpPropagationUtil.java]: CloudTraceFormat is not public so is not going to be in the doc. better to copy link and summary here. Here is a link https://cloud.google.com/trace/docs/support

File: contrib/http_util/src/main/java/io/opencensus/contrib/http/util/CloudTraceFormat.java [hunk 7]
```diff
@@ -58,7 +58,7 @@
  *   <li>"105445aa7843bc8bf206b120001000/123;o=0"
  * </ul>
  */
-final class CloudTraceFormat extends TextFormat {
+public final class CloudTraceFormat extends TextFormat {
```
Comment (bogdandrutu) [inline contrib/http_util/src/main/java/io/opencensus/contrib/http/util/CloudTraceFormat.java]: why public?

Comment (HailongWen) [reply contrib/http_util/src/main/java/io/opencensus/contrib/http/util/CloudTraceFormat.java]: I am trying not to duplicate the doc in CloudTraceFormat. I will revert this change back and adopt the solution you mentioned below.

**Pull request merged**